src/System.Device.Gpio/System/Device/Gpio/Drivers/LibgpiodDriver.cs

```diff
@@ -0,0 +1,195 @@
+using System.Text;
+using System.Threading;
+
+namespace System.Device.Gpio.Drivers
+{
+    public class LibgpiodDriver : GpioDriver
+    {
+        private const string GpioGet = "gpioget";
+        private const string GpioChip0 = "gpiochip0 ";
+        private const string GpioSet = "gpioset";
+        public int TimeoutMilliSeconds { get; set; } = (int)TimeSpan.FromSeconds(1).TotalMilliseconds;
+        private class MutableTuple<Item1, Item2>
+        {
+            public Item1 Value1 { get; set; }
+            public Item2 Value2 { get; set; }
+            public MutableTuple(Item1 item1, Item2 item2) {
+                Value1 = item1;
+                Value2 = item2;
+            }
+        }
+        private Dictionary<int, MutableTuple<PinMode, Process>> _pinNumberToPinModeAndProcess = new Dictionary<int, MutableTuple<PinMode, Process>>();
+        protected internal override int PinCount => throw new NotImplementedException();
+
+        protected internal override void AddCallbackForPinValueChangedEvent(int pinNumber, PinEventTypes eventType, PinChangeEventHandler callback)
+        {
+            throw new NotImplementedException();
+        }
+
+        protected internal override void ClosePin(int pinNumber)
+        {
+            MutableTuple<PinMode, Process> pinModeAndProcess;
+            if (_pinNumberToPinModeAndProcess.TryGetValue(pinNumber, out pinModeAndProcess)) 
+            {
+                Process existingProcess = pinModeAndProcess.Value2;
+                if (existingProcess != null) {
+                    existingProcess.Kill();
+                    existingProcess.WaitForExit(1);
+                    pinModeAndProcess.Value2 = null;
+                }
+                _pinNumberToPinModeAndProcess.Remove(pinNumber);
+            }
+        }
+
+        protected internal override int ConvertPinNumberToLogicalNumberingScheme(int pinNumber) => throw new PlatformNotSupportedException("This driver is generic so it can not perform conversions between pin numbering schemes.");
+
+        protected internal override PinMode GetPinMode(int pinNumber)
+        {
+            MutableTuple<PinMode, Process> pinModeAndProcess;
+            if (_pinNumberToPinModeAndProcess.TryGetValue(pinNumber, out pinModeAndProcess))
+             {
+                return pinModeAndProcess.Value1;
+            }
+            return PinMode.Input;
+        }
+
+        protected internal override bool IsPinModeSupported(int pinNumber, PinMode mode)
+        {
+            // Libgpiod command line tools does not support pull up or pull down resistors.
+            return mode != PinMode.InputPullDown && mode != PinMode.InputPullUp;
+        }
+
+        protected internal override void OpenPin(int pinNumber)
+        {
+            // do nothing
+        }
+
+        protected internal override PinValue Read(int pinNumber)
+        {
+            PinValue result = default(PinValue);
+            Process gpioGetCommand = new Process();
```

Comment:
Please use a `using` statement so the Process gets disposed.

Reply:
Oh, it looks like processes are held across functions. In that case the object as a whole needs to keep track of them and dispose them in its own Dispose() or some other scheme.

Reply:
Disposed here https://github.com/dotnet/iot/pull/96/files#diff-8d87a320a14f21bb1fe663cd68197b16R185 , not sure if its enough